packages/amqp/lib/AbstractAmqpBaseConsumer.ts

```diff
@@ -42,8 +42,24 @@ export abstract class AbstractAmqpBaseConsumer<MessagePayloadType extends object
     }
   }
 
+  /**
+   * Override to implement barrier pattern
+   */
+  public preHandlerBarrier(_message: MessagePayloadType, _messageType: string): Promise<boolean> {
+    return Promise.resolve(true)
```

Comment:
what if it ends with an error? probably we should return an Either here as well

Reply:
I am not sure if I am getting your message, do you mean it should return a `Promise<Either<'retryLater', 'success'>>`?

if it is the case, I have no strong opinion but I wanted to make it simpler with just a boolean (`true` means that the message can be processed and false means `retryLater`). If there is an exception it is going to be handled as we are actually handling them in the `internalProcessMessage` call -> 
```
const result: Either<'retryLater' | Error, 'success'> = await this.internalProcessMessage(
          deserializedMessage.result,
          messageType,
        )
          .catch((err) => {
            // ToDo we need sanity check to stop trying at some point, perhaps some kind of Redis counter
            // If we fail due to unknown reason, let's retry
            this.handleError(err)
            return {
              // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
              error: err,
            }
          })
          .finally(() => {
            this.transactionObservabilityManager?.stop(transactionSpanId)
          })
```

Reply:
gotcha, that should be good enough